src/contract_client.ts

```diff
@@ -0,0 +1,118 @@
+import { AssembledTransaction, MethodOptions } from './assembled_transaction'
+import { ContractSpec, xdr } from '.'
+
+export type XDR_BASE64 = string;
+
+export type ContractClientOptions = {
+  /**
+   * The account to use for signing transactions. If not provided, a null
+   * account will be used for the transaction simulation. If the transaction
+   * needs to be signed and sent, this is required.
+   */
+  publicKey?: string | Promise<string>;
```

Comment:
Can we get away with not making this optional? Like is there really any tangible benefit for a downstream user to be able to avoid passing a public key? They can just generate one if they are truly that lazy.